app/components/announcement_banner/announcement_banner.tsx

```diff
@@ -0,0 +1,187 @@
+
+const AnnouncementBanner = ({
+    bannerColor,
+    bannerDismissed,
+    bannerEnabled,
+    bannerText = '',
+    bannerTextColor = '#000',
+    allowDismissal,
+}: Props) => {
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+    const height = useSharedValue(0);
+    const theme = useTheme();
+    const [visible, setVisible] = useState(false);
+    const style = getStyle(theme);
+
+    const renderContent = useCallback(() => (
+        <ExpandedAnnouncementBanner
+            allowDismissal={allowDismissal}
+            bannerText={bannerText}
+        />
+    ), [allowDismissal, bannerText]);
+
+    const handlePress = useCallback(() => {
+        const title = intl.formatMessage({
+            id: 'mobile.announcement_banner.title',
+            defaultMessage: 'Announcement',
+        });
+
+        let snapPoint = SNAP_POINT_WITHOUT_DISMISS;
+        if (allowDismissal) {
+            snapPoint += DISMISS_BUTTON_HEIGHT;
+        }
+
+        bottomSheet({
+            closeButtonId: CLOSE_BUTTON_ID,
+            title,
+            renderContent,
+            snapPoints: [snapPoint, 10],
+            theme,
+        });
+    }, [theme.sidebarHeaderTextColor, intl.locale, renderContent, allowDismissal]);
+
+    const handleDismiss = useCallback(() => {
+        dismissAnnouncement(serverUrl, bannerText);
+    }, [serverUrl, bannerText]);
+
+    useEffect(() => {
+        const showBanner = bannerEnabled && !bannerDismissed && Boolean(bannerText);
+        setVisible(showBanner);
+    }, [bannerDismissed, bannerEnabled, bannerText]);
+
+    useEffect(() => {
+        height.value = withTiming(visible ? ANNOUNCEMENT_BAR_HEIGHT : 0, {
+            duration: 500,
+        });
+    }, [visible]);
+
+    const bannerStyle = useAnimatedStyle(() => ({
+        height: height.value,
+    }));
+
+    const bannerTextContainerStyle = useMemo(() => [style.bannerTextContainer, {
+        color: bannerTextColor,
+    }], [style, bannerTextColor]);
+
+    return (
+        <Animated.View
+            style={[style.background, bannerStyle]}
+        >
+            <View
+                style={[style.bannerContainer, {backgroundColor: bannerColor}]}
+            >
+                {visible &&
+                    <>
+                        <TouchableOpacity
+                            onPress={handlePress}
+                            style={style.wrapper}
+                        >
+                            <Text
+                                style={bannerTextContainerStyle}
+                                ellipsizeMode='tail'
+                                numberOfLines={1}
+                            >
+                                <CompassIcon
+                                    color={bannerTextColor}
+                                    name='information-outline'
+                                    size={18}
+                                />
+                                {'  '}
```

Comment:
is this line needed?

Reply:
Yes. This is an interesting issue.

In general, everything works in flexboxes. You can establish margins, paddings, etc.. For most scenarios, this works great, but when you start working with text, some things get a bit more complicated. In particular in this case, I was having trouble to correctly show the text centered.

So the solution was to use a `<Text>` element with center alignment. But I needed to center both the `i` icon and the text, so both came inside the `<Text>` component. And here where things get more interesting. Inside a `<Text>` component, you are no longer on a flex layout. You are on a text layout. And one of the implications that has is that you cannot use `margin` or `padding`. You can use it on the outermost `<Text>` but not on any element inside.

So... how can I make a separation between the `i` icon and the following text? White spaces. Therefore... if we remove this line, what we will have is that the text would be right next to the `i` icon, instead of separated a few pixels.